src/Compilers/CSharp/Portable/Emitter/EditAndContinue/EmitHelpers.cs

```diff
@@ -111,6 +119,37 @@ internal static class EmitHelpers
                 changedTypes: changedTypes.ToImmutableAndFree());
         }
 
+        private static bool ContainsAnonymousDelegates(
+            CSharpDefinitionMap definitionMap,
+            IReadOnlyDictionary<string, AnonymousTypeValue> previousDictionary,
+            IReadOnlyDictionary<string, AnonymousTypeValue> currentDictionary)
+        {
+            if (previousDictionary.Count == 0)
+            {
+                return true;
+            }
+
+            if (previousDictionary.Count > currentDictionary.Count)
```

Comment:
>if (previousDictionary.Count > currentDictionary.Count) [](http://example.com/codeflow?start=12&length=55)

Is there something wrong with removing the delegate?

Reply:
EnC relies on the current generation containing definitions for all types in previous generations, for mapping locals in the current stack frame from the previous generation for instance. If an anonymous delegate type definition has been removed (or the signature changed), we'd need to generate an equivalent type definition in the current generation and determining the delegate signature etc. is likely not worth the effort.